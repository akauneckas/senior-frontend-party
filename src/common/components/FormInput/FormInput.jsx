import React from 'react'
import PropTypes from 'prop-types';
import './FormInput.scss';

const propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    icon: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
    value: PropTypes.string,
};

const defaultProps = {
    value: '',
};

const FormInput = ({ id, name, type, placeholder, value, onChange, icon }) => (
    <div className="FormInput form-group">
        <div className="input-group">
            <div className="input-group-prepend">
                <span className="input-group-text">
                   <img alt={name} src={icon} />
                </span>
            </div>
            <input
                type={type}
                placeholder={placeholder}
                id={id}
                name={name}
                value={value}
                onChange={onChange}
                className="form-control"
            />
        </div>
    </div>
);

FormInput.propTypes = propTypes;
FormInput.defaultProps = defaultProps;

export default FormInput;
