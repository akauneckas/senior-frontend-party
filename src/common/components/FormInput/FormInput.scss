.FormInput {
  .input-group-text {
    background-color: #fff;
    padding-right: 0.375rem;
    border-right: none;
    padding-left: 1.5rem;
  }
  .form-control {
    border-left: none;
    padding-left: 0.375rem;
    line-height: 2.75;
    &:active {
      color: #999;
    }
  }
}