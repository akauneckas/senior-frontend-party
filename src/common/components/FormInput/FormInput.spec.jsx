import React from 'react';
import renderer from 'react-test-renderer';
import FormInput from './FormInput';
import IconUser from '../../../assets/icons/ico-user.png.png';

describe('<FormInput />', () => {
    it('should match snapshot', () => {
        const component = renderer.create(
            <FormInput
                type="text"
                id="username"
                name="username"
                placeholder="Username"
                value="foo"
                icon={IconUser}
                onChange={() => {}}
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
