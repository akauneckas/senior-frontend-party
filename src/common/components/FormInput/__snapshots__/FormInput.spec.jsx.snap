// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> should match snapshot 1`] = `
<div
  className="FormInput form-group"
>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
      >
        <img
          alt="username"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </span>
    </div>
    <input
      className="form-control"
      id="username"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value="foo"
    />
  </div>
</div>
`;
